File: examples/grep_speed_oxo.pl
use 5.030;

die "Usage: $0 FILENAME LIMIT" if @ARGV != 2;

my ($filename, $limit) = @ARGV;

open (my $fh, '<', $filename) or die;
while (my $line = <$fh>) {
    for (1..$limit) {
        print($line) if $line =~ /(.)y\1/;
    }
}

File: examples/grep_speed_oxo.py
import sys
import re

if len(sys.argv) != 3:
    exit(f"{sys.argv[0]} FILENAME LIMIT")

_, filename, limit = sys.argv

with open(filename) as fh:
    for line in fh:
        for _ in range(int(limit)):
            if re.search(r'(.)y\1', line):
                print(line)

